File: Examples/VisualBasic/ProgrammersGuide/Working-Shapes/Working-with-Icons-and-Pictures/ReplaceShapePicture.vb
Imports VisualBasic
Imports Aspose.Diagram
Imports System
Imports System.IO

Public Class ReplaceShapePicture
    Public Shared Sub Run()
        'ExStart:ReplaceShapePicture
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_Shapes()

        ' call a Diagram class constructor to load the VSD diagram
        Dim diagram As New Diagram(dataDir & Convert.ToString("ExtractAllImagesFromPage.vsd"))
        ' convert image into bytes array
        Dim imageBytes As Byte() = File.ReadAllBytes(dataDir & Convert.ToString("Picture.png"))

        'enter page index i.e. 0 for first one
        For Each shape As Shape In diagram.Pages(0).Shapes
            'Filter shapes by type Foreign
            If shape.Type = Aspose.Diagram.TypeValue.Foreign Then
                Using stream As New System.IO.MemoryStream(shape.ForeignData.Value)
                    'replace picture shape
                    shape.ForeignData.Value = imageBytes
                End Using
            End If
        Next

        ' save diagram
        diagram.Save(dataDir & Convert.ToString("ReplaceShapePicture_Out.vsdx"), SaveFileFormat.VSDX)
        'ExEnd:ReplaceShapePicture
    End Sub
End Class

File: Examples/CSharp/ProgrammersGuide/Working-Shapes/Working-with-Icons-and-Pictures/ReplaceShapePicture.cs
using Aspose.Diagram;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

namespace Aspose.Diagram.Examples.CSharp.ProgrammersGuide.Working_Shapes.Working_with_Icons_and_Pictures
{
    public class ReplaceShapePicture
    {
        public static void Run()
        {
            // ExStart:ReplaceShapePicture
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Shapes();

            // call a Diagram class constructor to load the VSD diagram
            Diagram diagram = new Diagram(dataDir + "ExtractAllImagesFromPage.vsd");
            // convert image into bytes array
            byte[] imageBytes = File.ReadAllBytes(dataDir + "Picture.png");

            // Enter page index i.e. 0 for first one
            foreach (Shape shape in diagram.Pages[0].Shapes)
            {
                // Filter shapes by type Foreign
                if (shape.Type == Aspose.Diagram.TypeValue.Foreign)
                {
                    using (System.IO.MemoryStream stream = new System.IO.MemoryStream(shape.ForeignData.Value))
                    {
                        // Replace picture shape
                        shape.ForeignData.Value = imageBytes;
                    }
                }
            }

            // save diagram
            diagram.Save(dataDir + "ReplaceShapePicture_Out.vsdx", SaveFileFormat.VSDX);
            // ExEnd:ReplaceShapePicture
        }
    }
}
